### DISCUSSION TITLE
Automatic Tangency For Arcs Via Shift Modifier

### DISCUSSION THREAD

**Discussion by TheMakerTales** [hlorus/CAD_Sketcher#208] [Feature Requests]
Currently, when you connect an arc to a point there is no way of making this connection tangent without then having to go in and select the two lines and then making them tangent to the arc. This can get VERY tedious especially as there is no chamfer/bevel tool a the moment. I know the chamfer/bevel tool is set as in issue to be developed but I think a powerful perk to the current arc creation tool would be an auto tangency system that is activated when holding down shift when the arc tool is active. 

This way when you are creating an arc you can hold down shift and connect it to one point and an auto tangent constraint would be created. If you would like the end of the arc to be tangent to the next connection feel free to hold shift or if you don't want it to be tangent let go of shift.

Here is a little gif "faking" the expected outcome.
![ezgif com-gif-maker (7)](https://user-images.githubusercontent.com/103122109/176423023-657b1958-a607-4b56-aa53-84d65ca687ef.gif)

**Comment by amrsoll** [comment]
That is interesting, and I don't think it would be super hard to implement.

Is the feature and shortcut consistent with other CADs?

**Reply by TheMakerTales** [reply]
I don't think any other CAD sketchers have this. Some have a multi-line tool like FreeCAD where you can cycle through different types of line connections all in one too : [See here](https://youtu.be/LIDak5PHO5I?list=PL6Fiih6ItYsWCE20KtUJYpiDPrCA2rVpN&t=352).

But naitevely freecad is like this:

![ezgif com-gif-maker (10)](https://user-images.githubusercontent.com/103122109/176635647-6e0a71dd-78c6-4e6b-b4e0-549ac139c541.gif)


As for fusion 360 the closest they have is this in their line tool: [https://youtu.be/Vb-Xf6N_9B0](https://youtu.be/Vb-Xf6N_9B0)

**Reply by amrsoll** [reply]
that is good to keep in mind! Perhaps we can keep in mind a more meta constraint shortcut property so it can work similarly to FreeCAD.

Example:
(Ctrl +)Shift + C while creating the new arc/line/curve to add the constraint.

**Comment by Intuos5** [comment]
Autocad's Polyline tool has such a thing as well. But, it would need to be in the other arc tools as well then, see #82 .
